src/gpu/intel/jit/emulation.hpp

```diff
@@ -591,7 +591,10 @@ struct EmulationImplementation {
         bool emulate64 = strategy.emulate64_mul;
 
         if (s0Q) {
-            stub();
+            const auto &temp
+                    = s1Signed ? state.temp[0].d() : state.temp[0].ud();
+            g.mov(1, temp, src1);
+            return emulInternal(g, mod, dst, temp, src0, strategy, state);
```

Comment:
To avoid a `mov` here, can you just swap `src0` and `src1` earlier (say at line 577) for 64 * 32 cases?

Reply:
No, the core issue is that the `S1` type may not be a `ngen::Register` (and the move is necessary for an ngen::Immediate operand), any attempt to swap  those values will just hit typing issues.

Reply:
Oh, right. Can you leave a TODO comment to improve performance on this path then?